src/common/offline_transformations/src/pruning/propagate_masks.cpp

```diff
@@ -1420,6 +1422,196 @@ class ngraph::pass::mask_propagation::Transpose : public MatcherPass {
     }
 };
 
+class ngraph::pass::mask_propagation::VariadicSplit : public MatcherPass {
+public:
+    VariadicSplit() {
+        auto input_pattern = pattern::any_input(pattern::has_static_rank());
+        auto axis_pattern = pattern::wrap_type<opset6::Constant>();
+        auto split_lengths_pattern = pattern::wrap_type<opset6::Constant>();
+        auto split_pattern =
+            pattern::wrap_type<opset6::VariadicSplit>({input_pattern, axis_pattern, split_lengths_pattern});
+
+        ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+            const auto& pattern_map = m.get_pattern_value_map();
+            auto axis_node = ngraph::as_type<opset6::Constant>(pattern_map.at(axis_pattern).get_node());
+            if (!axis_node)
+                return false;
+
+            const auto& input = pattern_map.at(input_pattern);
+            const auto input_mask = getMask(input);
+            if (!input_mask)
+                return false;
+
+            auto axis = axis_node->cast_vector<int64_t>()[0];
+            if (axis < 0)
+                axis += input_mask->size();
+
+            auto split = pattern_map.at(split_pattern).get_node();
+
+            auto split_lengths_const =
+                ngraph::as_type<opset6::Constant>(pattern_map.at(split_lengths_pattern).get_node());
+            if (!split_lengths_const)
+                return false;
+            auto split_lengths = split_lengths_const->cast_vector<int64_t>();
```

Comment:
```suggestion
            const auto split_lengths = split_lengths_const->cast_vector<int64_t>();
```

Reply:
It cannot be const since it's modified later.